daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TransactionBuilder.scala

```diff
@@ -41,13 +41,19 @@ final class TransactionBuilder(pkgTxVersion: Ref.PackageId => TransactionVersion
   def add(node: Node, parent: NodeId): NodeId = ids.synchronized {
     lazy val nodeId = newNode(node) // lazy to avoid getting the next id if the method later throws
     nodes = nodes.mapResult { ns =>
-      val exercise = ns(parent) match {
+      val VersionedNode(version, node) = ns(parent)
+      val exercise = node match {
         case exe: TxExercise => exe
         case _ =>
           throw new IllegalArgumentException(
             s"Node ${parent.index} either does not exist or is not an exercise")
       }
-      ns.updated(parent, exercise.copy(children = exercise.children.slowSnoc(nodeId)))
+      ns.updated(
+        parent,
+        VersionedNode(
+          version,
+          exercise.copy(children = exercise.children.slowSnoc(nodeId))
```

Comment:
Probably for another PR, but is `slowSnoc` a good idea?

Reply:
It is probably a bad idea. I will take care of that in another PR.